collector-bitmovin-player/src/main/java/com/bitmovin/analytics/bitmovin/player/BitmovinSdkAdapter.java

```diff
@@ -454,42 +477,17 @@ public void onError(ErrorEvent errorEvent) {
             ErrorCode errorCode = exceptionMapper.map(errorEvent);
 
             stateMachine.setErrorCode(errorCode);
-            if (!isVideoPlayed && isVideoAttemptedPlay) {
-                videoStartTimeout.cancel();
+            if (!stateMachine.isStartupFinished() && isVideoAttemptedPlay) {
                 stateMachine.setVideoStartFailedReason(VideoStartFailedReason.PLAYER_ERROR);
             }
             stateMachine.transitionState(PlayerState.ERROR, videoTime);
         }
     };
 
-    private OnReadyListener onReadyListener = new OnReadyListener() {
-        @Override
-        public void onReady(ReadyEvent readyEvent) {
-            playerIsReady = true;
-            Log.d(TAG, "On Ready Listener");
-
-            if (bitmovinPlayer.isPlaying()) {
-                stateMachine.transitionState(PlayerState.PLAYING, getPosition());
-                // autoplay
-                isVideoAttemptedPlay = true;
-                videoStartTimeout.start();
-            } else {
-                stateMachine.transitionState(PlayerState.PAUSE, getPosition());
-            }
-
-        }
-    };
-
-    private CountDownTimer videoStartTimeout = new CountDownTimer(Util.VIDEOSTART_TIMEOUT, 1000) {
-        @Override
-        public void onTick(long millisUntilFinished) {
-        }
-
+    private OnAdBreakStartedListener onAdBreakStartedListener = new OnAdBreakStartedListener() {
         @Override
-        public void onFinish() {
-            Log.d(TAG, "VideoStartTimeout finish");
-            stateMachine.setVideoStartFailedReason(VideoStartFailedReason.TIMEOUT);
-            stateMachine.transitionState(PlayerState.EXITBEFOREVIDEOSTART, getPosition());
+        public void onAdBreakStarted(AdBreakStartedEvent adBreakStartedEvent) {
+            stateMachine.startAd(getPosition());
```

Comment:
How do you get out of the ad state again? Looks like you track all the play/pause etc during an ad as well.

Reply:
When the ad has finished there is a state transition flow like this
Ad playing
onPlay
AD -> Startup
onPlaying
Startup -> Playing

Do you see issues here?
No Play event listening during ads?